File: studio/backend/api/v1/schemas/jobs.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel


class JobResponse(BaseModel):
    job_id: UUID
    report_id: UUID
    status: str
    current_agent: str | None = None
    progress_pct: int
    error_message: str | None = None
    created_at: datetime

File: studio/frontend/src/lib/api/jobs.ts
import { apiRequest } from "@/lib/api/client";
import type { ReportStatus } from "@/lib/api/reports";

export interface JobStatusResponse {
  job_id: string;
  report_id: string;
  status: ReportStatus;
  current_agent: string | null;
  progress_pct: number;
  error_message: string | null;
  created_at: string;
}

export const getJob = async (
  token: string,
  jobId: string,
): Promise<JobStatusResponse> => apiRequest<JobStatusResponse>(`/jobs/${jobId}`, token);
